Flaky integration test: `test_failing_plugin`

Nodeid: `tests_integration/plugin/test_failing_plugin.py::test_failing_plugin`
# 🟢 Setup passed
# 🔴 Call failed
```python
runner = <tests_integration.conftest.SnowCLIRunner object at 0x7f1818ee61a0>
test_root_path = PosixPath('/home/runner/work/snowflake-cli/snowflake-cli/tests_integration')
_install_plugin = None
caplog = <_pytest.logging.LogCaptureFixture object at 0x7f1818ee6b60>

    @pytest.mark.integration
    def test_failing_plugin(runner, test_root_path, _install_plugin, caplog):
        config_path = (
            test_root_path / "config" / "plugin_tests" / "failing_plugin_config.toml"
        )
    
        result = runner.invoke(["--config-file", config_path, "connection", "list"])
    
>       assert (
            "Cannot register plugin [failing_plugin]: Some error in plugin"
            in caplog.messages
        )
E       AssertionError: assert 'Cannot register plugin [failing_plugin]: Some error in plugin' in ['Creating directory /tmp/tmpfrjsoet0/.snowflake/logs']
E        +  where ['Creating directory /tmp/tmpfrjsoet0/.snowflake/logs'] = <_pytest.logging.LogCaptureFixture object at 0x7f1818ee6b60>.messages

tests_integration/plugin/test_failing_plugin.py:28: AssertionError
```
https://github.com/snowflakedb/snowflake-cli/blob/4c2eabf73e8b2b5c764fdc8395b39e51126f05b1/tests_integration/plugin/test_failing_plugin.py#L28
# 🟢 Teardown passed

--- a/src/snowflake/cli/_app/cli_app.py
+++ b/src/snowflake/cli/_app/cli_app.py
@@ -256,7 +256,6 @@ class CliAppFactory:
                 "--commands-registration",
                 help="Commands registration",
                 hidden=True,
-                is_eager=True,
                 callback=self._commands_registration_callback(),
             ),
         ) -> None:

